react/features/analytics/middleware.js

```diff
@@ -79,13 +79,28 @@ function calculateLocalTrackDuration(state) {
  * @returns {Function}
  */
 MiddlewareRegistry.register(store => next => action => {
-    if (action.type === SET_CONFIG) {
+    switch (action.type) {
+    case SET_CONFIG:
         if (navigator.product === 'ReactNative') {
             // Reseting the analytics is currently not needed for web because
             // the user will be redirected to another page and new instance of
             // Analytics will be created and initialized.
             resetAnalytics();
         }
+        break;
+    case SET_ROOM: {
+        // createHandlers is called before the SET_ROOM action is executed in order for Amplitude to initialize before
```

Comment:
This sounds a bit risky. Couldn't we lose events at the beginning of the meeting, if it takes too long for the handlers to be created? Is that a risk we are willing to assume?

Reply:
1) Actually with this change the handlers are created earlier than before. Or at least this was my intention here. Did I miss something?
2) If I remember correctly we are caching all events before we set the handlers and once the handlers are set we sending them.

Reply:
Ah, correct!